File: src/main/java/com/hy/frame/bean/ResultInfo.java
package com.hy.frame.bean;

import java.util.HashMap;
import java.util.Map;

/**
 * 服务器返回数据
 * Created by HeYan on 2015/10/17.
 */
public class ResultInfo {
    /**
     * 本地错误 默认
     */
    public static final int CODE_ERROR_DEFAULT = -250;
    /**
     * 网络错误
     */
    public static final int CODE_ERROR_NET = -251;
    /**
     * 解析错误
     */
    public static final int CODE_ERROR_DECODE = -252;
    private int requestCode;//请求码，接口编号
    private long qid;//队列ID
    private Object obj;//返回结果
    private int errorCode;//错误码
    private String msg;//描述
    private int requestType;
    private Map<String, String> maps;//其他

    /**
     * 请求码，接口编号
     */
    public int getRequestCode() {
        return requestCode;
    }

    /**
     * 请求码，接口编号
     */
    public void setRequestCode(int requestCode) {
        this.requestCode = requestCode;
    }

    /**
     * 队列ID
     */
    public long getQid() {
        return qid;
    }

    /**
     * 队列ID
     */
    public void setQid(long qid) {
        this.qid = qid;
    }

    /**
     * 返回结果
     */
    public <T> T getObj() {
        return (T) obj;
    }

    /**
     * 返回结果
     */
    public void setObj(Object obj) {
        this.obj = obj;
    }

    /**
     * 错误码 250 本地错误 251 网络错误
     */
    public int getErrorCode() {
        return errorCode;
    }

    /**
     * 错误码
     */
    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    /**
     * 描述
     */
    public String getMsg() {
        return msg;
    }

    /**
     * 描述
     */
    public void setMsg(String msg) {
        this.msg = msg;
    }

    public int getRequestType() {
        return requestType;
    }

    public void setRequestType(int requestType) {
        this.requestType = requestType;
    }

    public void putValue(String key, String value) {
        if (maps == null)
            maps = new HashMap<>();
        maps.put(key, value);
    }

    public String getValue(String key) {
        if (maps != null) return maps.get(key);
        return null;
    }
}

File: src/main/java/com/hy/frame/bean/ResultInfo.kt
package com.hy.frame.bean

import java.util.*

/**
 * ResultInfo 服务器返回数据
 * @author HeYan
 * @time 2017/5/23 10:51
 */
class ResultInfo {
    /**
     * 请求码，接口编号
     */
    /**
     * 请求码，接口编号
     */
    var requestCode: Int = 0//请求码，接口编号
    /**
     * 队列ID
     */
    /**
     * 队列ID
     */
    var qid: Long = 0//队列ID
    private var obj: Any? = null//返回结果
    /**
     * 错误码 250 本地错误 251 网络错误
     */
    /**
     * 错误码
     */
    var errorCode: Int = 0//错误码
    /**
     * 描述
     */
    /**
     * 描述
     */
    var msg: String? = null//描述
    var requestType: Int = 0
    private var maps: MutableMap<String, String>? = null//其他


    /**
     * 返回结果
     */
    @Suppress("UNCHECKED_CAST")
    fun <T> getObj(): T {
        return obj as T
    }

    /**
     * 返回结果
     */
    fun setObj(obj: Any?) {
        this.obj = obj
    }

    fun putValue(key: String, value: String) {
        if (maps == null)
            maps = HashMap<String, String>()
        maps!!.put(key, value)
    }

    fun getValue(key: String): String? {
        if (maps != null) return maps!![key]
        return null
    }

    fun getMaps(): MutableMap<String, String>? = maps

    companion object {
        /**
         * 本地错误 默认
         */
        val CODE_ERROR_DEFAULT = -250
        /**
         * 网络错误
         */
        val CODE_ERROR_NET = -251
        /**
         * 解析错误
         */
        val CODE_ERROR_DECODE = -252
    }
}
